src/main/java/edu/ucsb/mapache/controllers/SearchHistoryController.java

```diff
@@ -63,20 +63,28 @@ public DecodedJWT getJWT(String authorization) {
         if (!authControllerAdvice.getIsMember(authorization))
             return getUnauthorizedResponse("member");
         
-       if (!authControllerAdvice.getIsAdmin(authorization)) {
+      
+        
+        Iterable<UserSearch> usersearch = usersearchRepository.findAll();
+        String body = mapper.writeValueAsString(usersearch);
+        return ResponseEntity.ok().body(body);
+    }
+    
+    @GetMapping("/specificuser")
+    public ResponseEntity<String> getspecificSearches(@RequestHeader("Authorization") String authorization)
+            throws JsonProcessingException {
+       
+        
+      
```

Comment:
Additionally, this endpoint should be restricted to only members and admins as well.

Reply:
Add conditional statement to check for membership.